### ISSUE TITLE
Google Pub/Sub gRPC Exception when publishing to PubSub

### ISSUE THREAD

**Issue by christophermiliotis** [akka/alpakka#2237]
### Versions used 

Akka version: 2.6.1
Alpakka version: 2.0.0-RC1

### Expected Behavior

Publishing to pub/sub should succeed.

### Actual Behavior

Upon upgrading alpakka from v2.0.0-M2 to 2.0.0-RC1, I noticed that our integration tests (executed against a real PubSub) started failing.

In particular, executing this:
```
val request = PublishRequest().withTopic(topic.name).addAllMessages(pubSubMsgs)
GrpcPublisherExt().publisher.client.publish(request)
```

results in the following exception:
```
i.g.StatusRuntimeException: UNIMPLEMENTED: HTTP status code 404
```

### Relevant logs

Stack trace
```
stack_trace: 
io.grpc.StatusRuntimeException: UNIMPLEMENTED: HTTP status code 404
invalid content-type: text/html; charset=UTF-8
headers: Metadata(:status=404,content-type=text/html; charset=UTF-8,referrer-policy=no-referrer,content-length=1602,date=Fri, 27 Mar 2020 11:11:34 GMT)
DATA-----------------------------
<!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 404 (Not Found)!!1</title>
  <p><b>404.</b> <ins>That’s an error.</ins>
  <p>The requested URL <code>/google.pubsub.v1.Subscriber/StreamingPull</code> was not found on this server.  <ins>That’s all we know.</ins>

	at io.grpc.Status.asRuntimeException(Status.java:533)
	at akka.grpc.internal.AkkaNettyGrpcClientGraphStage$$anon$1.onCallClosed(AkkaNettyGrpcClientGraphStage.scala:165)
	at akka.grpc.internal.AkkaNettyGrpcClientGraphStage$$anon$1.$anonfun$callback$1(AkkaNettyGrpcClientGraphStage.scala:71)
	at akka.grpc.internal.AkkaNettyGrpcClientGraphStage$$anon$1.$anonfun$callback$1$adapted(AkkaNettyGrpcClientGraphStage.scala:67)
	at akka.stream.impl.fusing.GraphInterpreter.runAsyncInput(GraphInterpreter.scala:466)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:497)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:599)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:768)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:783)
	at akka.actor.Actor.aroundReceive(Actor.scala:533)
	at akka.actor.Actor.aroundReceive$(Actor.scala:531)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:690)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:573)
	at akka.actor.ActorCell.invoke(ActorCell.scala:543)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:269)
	at akka.dispatch.Mailbox.run(Mailbox.scala:230)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:242)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
"   
  timestamp: "2020-03-27T11:11:34.289+00:00"   
 }
" 
```

### Reproducible Test Case

This should be reproducible against a real Pub/Sub resource.
Please note, the issue does not manifest if tested against a mock impl of PubSub - our acceptance tests run with PubSub emulator and they are passing.

**Comment by ennru** [comment]
Thank you for reporting.
The major change in Alpakka 2.0.0-RC1 is the use of Akka gRPC 0.8.0.

@raboof Any idea?

**Comment by seglo** [comment]
@christophermiliotis Do previous versions of Alpakka work?  Could this just be a transient connection issue?

**Comment by christophermiliotis** [comment]
@seglo Yes, they do, with akka gRPC 0.7.2

In fact, here's a list of the most relevant dependencies with which the integration tests are passing:
```
val akkaVersion = "2.6.1"
lazy val alpakkaVersion = "2.0.0-M2"
lazy val akkaStreamKafkaVersion = "2.0.0-RC1"

lazy val akkaDependencies = Seq(
    "akka-actor",
    "akka-stream",
    "akka-slf4j",
    "akka-discovery"
  ).map("com.typesafe.akka" %% _ % akkaVersion)

lazy val alpakkaDependencies = Seq(
    "akka-stream-alpakka-ftp",
    "akka-stream-alpakka-csv",
    "akka-stream-alpakka-google-cloud-pub-sub-grpc",
    "akka-stream-alpakka-json-streaming",
    "akka-stream-alpakka-google-cloud-storage",
    "akka-stream-alpakka-s3"
  ).map("com.lightbend.akka" %% _ % alpakkaVersion)
    .map(_.exclude("com.google.api.grpc", "proto-google-cloud-pubsub-v1")
          .exclude("com.fasterxml.jackson.core", "jackson-databind")) ++ // Removing the transitive dependency from jackson-databind v2.10.0, implied by Alpakka
    Seq(
      "com.fasterxml.jackson.core" % "jackson-databind" % "2.9.8", // Forcing jackson-databind to v2.9.8 because scio requires *that* version explicitly
      "org.conscrypt"     % "conscrypt-openjdk-uber" % "1.4.2",
      "com.typesafe.akka" %% "akka-stream-kafka"     % akkaStreamKafkaVersion
    )

lazy val grpcAndProtobufDependencies = Seq(
    "io.grpc" % "grpc-netty" % scalapb.compiler.Version.grpcJavaVersion,
    "com.thesamet.scalapb" %% "scalapb-runtime-grpc" % scalapb.compiler.Version.scalapbVersion,
    "com.thesamet.scalapb" %% "scalapb-runtime" % scalapb.compiler.Version.scalapbVersion % "protobuf"
  )
```

**Comment by lustefaniak** [comment]
I have a feeling this might be related to https://github.com/akka/alpakka/issues/2232 - maybe some `Host` header is not sent anymore and some generic load balancer handles the request?

**Comment by raboof** [comment]
Yes, likely related to https://github.com/akka/akka-grpc/issues/898

**Comment by raboof** [comment]
@lustefaniak  I suspect this might be fixed in snapshot `0.8.2+5-c7e7794a` of akka-grpc.

Could you test this by adding `resolvers += Resolver.bintrayRepo("akka", "maven")` to your sbt config and adding an explicit dependency on `"com.lightbend.akka.grpc" % akka-grpc-runtime" %% "0.8.2+5-c7e7794a"`?

**Comment by christophermiliotis** [comment]
Thanks @raboof , I will give that a go and report back 👍

**Comment by ennru** [comment]
Fixed with #2259
